internal/state/memory.go

```diff
@@ -4,62 +4,83 @@ import (
 	"context"
 	"fmt"
 	"sync"
+
+	"github.com/base-org/pessimism/internal/core"
 )
 
+/*
+	NOTE - This is a temporary implementation of the state store.
+*/
+
 // stateStore ... In memory state store
 type stateStore struct {
-	store map[string][]string
+	sliceStore map[string][]string
 
 	sync.RWMutex
 }
 
 // NewMemState ... Initializer
 func NewMemState() Store {
```

Comment:
Maybe worth doing a memory usage test depending on a max number of data that a block might hold that pertinent to us ?